src/materialized/bin/materialized.rs

```diff
@@ -155,6 +160,15 @@ fn run() -> Result<(), failure::Error> {
     let processes = popts.opt_get_default("processes", 1)?;
     let address_file = popts.opt_str("address-file");
     let gather_metrics = !popts.opt_present("no-prometheus");
+    let debug_allowed = popts.opt_present("debug");
```

Comment:
And then here I believe you can simplify to:

```rust
if cfg!(debug_assertions) && !pops.opt_present(debug) {
  bail!("cannot run debug build without --debug option")
}
```

Reply:
My only code comment is that I think an error message like `cannot run...` is clearer.